app/auth/authenticate_user.rb

```diff
@@ -12,18 +11,21 @@ def initialize(name, password)
 	# Then add a method #token to get the access_token
 	# (You can also use an attr_reader here and set the token on successful auth)
 	def call
-		JsonWebToken.encode(user_id: user.id) if user
+		if auth_user
+      @token = JsonWebToken.encode(user_id: @user.id)
+    else
+      errors.add(:user_authentication, 'Invalid credentials')
+    end
 	end
 
 	private
 
-	def user
-		user = User.find_by_name(name)
-    if user && user.authenticate(password)
-      return user
+  def auth_user
+    @user = User.find_by_name(@name)
+    if @user && @user.authenticate(@password)
+      return true
     else
-      user.errors.add :user_authentication, 'Invalid credentials'
-      nil
+      return false
     end
```

Comment:
This is basically ok, but the code looks better if you use the check if the user exists as a guard clause and then just return the result of the authenticate:

        @user = User.find_by_name(@name)
        return false unless @user
        return @user.authenticate(@password)

Generally, try to use as few nestings and lines as possible